### ISSUE TITLE
Reproducing issues

### ISSUE THREAD

**Issue by jaclyngu** [CaraJ7/T2I-R1#14]
Hi, thanks for the great work! I'm trying to reproduce the training on 8 H100s, but the default num_generations=8 in the run_grpo.sh config would cause OOM, so I changed it to 4. The attached figure shows the training curves. Among the four reward models, only HPSv2 reward is improving. So I'm wondering:
1. how to train it on H100s (80GB) with num_generations=8?
2. Do my training curves look right? It'll be greatly appreciated if you could share more info about the training statistics!
Many thanks in advance!

<img width="1340" height="668" alt="Image" src="https://github.com/user-attachments/assets/63c7af32-cd36-4bce-9254-8356b131e352" />

**Comment by johnsonkao0213** [comment]
My reward curve also looks same as you and training loss did not decrease, I also use H100s to run, not sure whether the result is correct since I did not change any lines of code?

**Comment by CaraJ7** [comment]
Hi @jaclyngu and @johnsonkao0213,

The reward curve behaviors you're observing are expected and align with our training procedure.

Regarding the training loss increase: this is normal when using pure on-policy updates. In our implementation, we perform only one iteration per GRPO cycle with μ=1 (as described in [the paper](https://arxiv.org/pdf/2402.03300) and our code [here](https://github.com/CaraJ7/T2I-R1/blob/941b6a9cc85a072391faf0d3416e4bc2bcd463bb/src/t2i-r1/src/open_r1/trainer/grpo_trainer.py#L767)). Under these conditions, the value of training loss is exactly the KL divergence between the policy model and the reference model, which naturally increases as the policy diverges from the reference during training.
